Client/Frontend/Browser/New Tab Page/NewTabPageViewController.swift

```diff
@@ -121,30 +126,57 @@ class NewTabPageViewController: UIViewController, Themeable {
             }),
             DuckDuckGoCalloutSectionProvider(profile: profile, action: { [weak self] in
                 self?.delegate?.tappedDuckDuckGoCallout()
-            }),
+            })
         ]
+        
+        if !PrivateBrowsingManager.shared.isPrivateBrowsing {
+            sections.append(
+                BraveTodaySectionProvider(
+                    dataSource: feedDataSource,
+                    actionHandler: { [weak self] in
+                        self?.handleBraveTodayAction($0)
+                    }
+                )
+            )
+            layout.braveTodaySection = sections.firstIndex(where: { $0 is BraveTodaySectionProvider })
+        }
+
         collectionView.delegate = self
         collectionView.dataSource = self
         applyTheme(Theme.of(tab))
         
         background.changed = { [weak self] in
             self?.setupBackgroundImage()
         }
+        
+        Preferences.BraveToday.isEnabled.observe(from: self)
+        feedDataSource.observeState(from: self) { [weak self] in
+            self?.handleFeedStateChange($0, $1)
+        }
+        NotificationCenter.default.addObserver(self, selector: #selector(checkForUpdatedFeed), name: UIApplication.didBecomeActiveNotification, object: nil)
```

Comment:
Shouldn't the updates be checked only if the user is in regular browsing mode?

Reply:
`checkForUpdatedFeed` does pre-checks for whether or not Brave Today is enabled and they are not private browsing before actually making any network requests.